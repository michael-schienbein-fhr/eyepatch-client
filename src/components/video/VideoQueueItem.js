import { useEffect, useState } from 'react';
import './video-item.css';

const VideoQueueItem = ({ video, handleVideoRemoval, handleVideoChange, globalVideoId }) => {
  const [activeVideo, setActiveVideo] = useState(false);

  useEffect(() => {
    if (globalVideoId === video.videoId) {
      setActiveVideo(true);
    } else {
      setActiveVideo(false);
    }
  }, [globalVideoId, video.videoId])
  return (
    <>
      <div className='video-item-container' onClick={() => handleVideoChange(video, 'queue')} className='video-item item'>
        <div className="video-item-remove" onClick={() => handleVideoRemoval(video)}>X</div>
        <img className={activeVideo ? 'image active' : 'image'} src={video.thumbnail} alt={video.description} />
        <span>{video.title}</span>
      </div>
    </>
  )
};
export default VideoQueueItem;
