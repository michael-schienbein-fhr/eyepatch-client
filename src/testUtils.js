import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const demoQueueVideo = {
  username: 'testuser2',
  type: 'video',
  action: 'add',
  text: '"testVideoTitle" added to queue for user: "testuser2".',
  videoId: 'testVideoId',
  title: 'testVideoTitle',
  description: 'testDescription',
  thumbnail: 'testThumbnail'
}
const demoSearchVideo = { "kind": "youtube#searchResult", "etag": "3_5Vn-wWRzIN5QC7aOEWvZsmMwI", "id": { "kind": "youtube#video", "videoId": "GjbibR6DV1s" }, "snippet": { "publishedAt": "2015-08-11T17:03:57Z", "channelId": "UCi_ZH3bYHidyHsETH5hbTFQ", "title": "Dusky - Skin Deep (Official Video)", "description": "Dusky's EP 'Ordinary World' is out now. iTunes:http://po.st/DuskyowEP Spotify: http://po.st/DuskyOWSp Beatport: http://po.st/OrdinaryWorldEP \"This EP was all ...", "thumbnails": { "default": { "url": "https://i.ytimg.com/vi/GjbibR6DV1s/default.jpg", "width": 120, "height": 90 }, "medium": { "url": "https://i.ytimg.com/vi/GjbibR6DV1s/mqdefault.jpg", "width": 320, "height": 180 }, "high": { "url": "https://i.ytimg.com/vi/GjbibR6DV1s/hqdefault.jpg", "width": 480, "height": 360 } }, "channelTitle": "DuskyMusic", "liveBroadcastContent": "none", "publishTime": "2015-08-11T17:03:57Z" } }

const testRoom = {
  "id": 1,
  "roomOwner": "testadmin",
  "roomName": "testadmin",
  "roomMembers": null,
  "videoQueue": null,
  "hasPass": true,
  "created_at": "2021-10-22T10:34:59.443Z"
}
const UserProvider =
  ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
  );

export { UserProvider, demoUser, demoQueueVideo, demoSearchVideo, testRoom };

