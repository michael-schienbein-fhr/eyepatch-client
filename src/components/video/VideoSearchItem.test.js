import { render } from "@testing-library/react";
import userEvent from '@testing-library/user-event'
import { UserProvider, demoQueueVideo, demoSearchVideo } from "../../testUtils";
import VideoQueueItem from "./VideoQueueItem";

// const BASE_URL = process.env.REACT_APP_BASE_URL || "http://localhost:3001";
// const nock = require('nock')



it("matches snapshot", function () {
  window.HTMLElement.prototype.scrollIntoView = function () { };
  const { asFragment } = render(
    <UserProvider>
      <VideoQueueItem
        video={demoSearchVideo}
      />
    </UserProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
