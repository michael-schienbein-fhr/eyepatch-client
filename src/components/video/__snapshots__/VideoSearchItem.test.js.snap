// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="video-item item"
  >
    <div
      class="video-item-remove"
    >
      X
    </div>
    <img
      class="image active"
    />
    <span />
  </div>
</DocumentFragment>
`;
